modin/core/storage_formats/pandas/parsers.py

```diff
@@ -379,6 +379,37 @@ def parse(fname, **kwargs):
         return _split_result_for_readers(1, num_splits, df) + [length, width]
 
 
+@doc(_doc_pandas_parser_class, data_type="custom text")
+class CustomTextExperimentalParser(PandasParser):
+    @staticmethod
+    @doc(_doc_parse_func, parameters=_doc_parse_parameters_common)
+    def parse(fname, **kwargs):
+        num_splits = kwargs.pop("num_splits", None)
+        start = kwargs.pop("start", None)
+        end = kwargs.pop("end", None)
+        columns = kwargs.pop("columns", None)
+        custom_parser = kwargs.pop("custom_parser", None)
+
+        assert start is not None and end is not None
+        assert columns is not None and custom_parser is not None
+
+        # pop "compression" from kwargs because bio is uncompressed
+        with OpenFile(fname, "rb", kwargs.pop("compression", "infer")) as bio:
+            bio.seek(start)
+            to_read = bio.read(end - start)
```

Comment:
We need to limit the number of bytes that can be read. I did not find how to do it with built-in IO tools/classes. I think this can be done - by creating `BufferedReader` own class, which will keep track of the number of bytes read. Do you know an easier solution?

Reply:
why we cannot just pass the `bio` handle to the reader?

Reply:
If we do this, then in each process reader will read bytes from `start` position to the end of the file, instead of reading up to `end` position.